### PR TITLE
Make Bulk Upload Compatible with CASS' .xlsx Format

### PR DESCRIPTION
**Pull request by Daniel-Jang**
Original Issue: https://github.com/cass-degrees/CASS-Degrees-Code/issues/217

Added support for Excel sheets, either in the custom format that the bulk uploader could already support, or in the format CASS uses.

To test this branch out, try the following file: https://drive.google.com/open?id=1EdQHhrNSd2QhO4c2A_su3x3K7hHQKQoI

Note that not all will be successfully added, but most will be.

### CODE REVIEW

**Review by JackSekoranja — CHANGES_REQUESTED** [review]
Functionally it works well, but if an incorrect excel file is chosen the website will crash, so it would be good to make it more robust regarding incorrect xlsx files. I also think it is unnecessary to display courses that didn't upload due to already being in the system, especially given the large number of courses that failed.

This also may be too difficult (so take this feedback as optional) but since there are so many courses on the error page, it might be useful to be able to hide some of them somehow.

**Review by JackSekoranja — APPROVED** [review]
Works, would like you to prevent duplicate courses from printing but otherwise good

File: cassdegrees/ui/views/bulk_data_upload.py [hunk 1]
```diff
@@ -19,10 +20,15 @@
 # ARTI-SPEC%2016%Artificial Intelligence%24%SPEC
 # ...
 
+# 20/08/19 changes: Support added for excel sheets in the same rules as above (no % needed, but put each word between
+# the % sign in a new cell in that row.
+
+# 21/08/19 changes: Support added for CASS teaching plan excel file.
```
Comment (j-selby) [inline cassdegrees/ui/views/bulk_data_upload.py]: Shouldn't this be left to commits?

**Review by j-selby — APPROVED** [review]
Looks good!

**Pull request merged**